graphql/resolvers.go

```diff
@@ -1295,42 +1268,92 @@ func (r *queryResolver) TaskTests(ctx context.Context, taskID string, execution
+	if dbTask == nil || err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("cannot find task with id %s", taskID))
+	}
 	baseTask, err := dbTask.FindTaskOnBaseCommit()
 	if err != nil {
 		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error finding base task for task %s: %s", taskID, err))
 	}
+
 	baseTestStatusMap := make(map[string]string)
-	if cedarTestResults == nil {
-		var taskExecution int
-		taskExecution = dbTask.Execution
-
-		if baseTask != nil {
-			baseTestResults, _ := r.sc.FindTestsByTaskId(data.FindTestsByTaskIdOpts{TaskID: baseTask.Id, Execution: taskExecution})
-			for _, t := range baseTestResults {
-				baseTestStatusMap[t.TestFile] = t.Status
-			}
+	// Fetch base tests and populate baseTestStatusMap
+	if baseTask != nil {
+		baseTaskLatestExec, err := task.GetLatestExecution(baseTask.Id)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error getting getting latest execution for task %s: %s", baseTask.Id, err.Error()))
 		}
 
-		paginatedFilteredTests, err := r.sc.FindTestsByTaskId(data.FindTestsByTaskIdOpts{
-			TaskID:    taskID,
-			TestName:  testNameParam,
-			Statuses:  statusesParam,
-			SortBy:    sortBy,
-			GroupID:   groupIdParam,
-			SortDir:   sortDir,
-			Limit:     limitParam,
-			Execution: taskExecution,
-			Page:      pageParam,
-		})
+		baseTestResultOpts := apimodels.GetCedarTestResultsOptions{
+			BaseURL:   evergreen.GetEnvironment().Settings().Cedar.BaseURL,
+			Execution: baseTaskLatestExec,
+		}
 
+		if len(baseTask.ExecutionTasks) > 0 {
+			baseTestResultOpts.DisplayTaskID = baseTask.Id
+		} else {
+			baseTestResultOpts.TaskID = baseTask.Id
+		}
+
+		cedarBaseTestResults, err := apimodels.GetCedarTestResults(ctx, baseTestResultOpts)
 		if err != nil {
-			return nil, ResourceNotFound.Send(ctx, err.Error())
+			grip.Warning(message.WrapError(err, message.Fields{
+				"task_id": baseTask.Id,
+				"message": "problem getting cedar test results",
+			}))
 		}
-		for _, t := range paginatedFilteredTests {
+
+		if cedarBaseTestResults != nil && len(cedarBaseTestResults) > 0 {
+			for _, t := range cedarBaseTestResults {
+				baseTestStatusMap[t.DisplayTestName] = t.Status
+			}
+		} else {
+			baseTestResults, _ := r.sc.FindTestsByTaskId(data.FindTestsByTaskIdOpts{TaskID: baseTask.Id, Execution: baseTaskLatestExec})
+			if baseTestResults != nil && len(baseTestResults) > 0 {
+				for _, t := range baseTestResults {
+					baseTestStatusMap[t.TestFile] = t.Status
+				}
+			}
+		}
+	}
+
+	// Fetch tests and link base test status.
+	opts := apimodels.GetCedarTestResultsOptions{
+		BaseURL:   evergreen.GetEnvironment().Settings().Cedar.BaseURL,
+		Execution: dbTask.Execution,
+	}
+	if len(dbTask.ExecutionTasks) > 0 {
+		opts.DisplayTaskID = taskID
+	} else {
+		opts.TaskID = taskID
+	}
+
+	cedarTestResults, err := apimodels.GetCedarTestResults(ctx, opts)
+
+	if err != nil {
+		grip.Warning(message.WrapError(err, message.Fields{
```

Comment:
this one too I would think should be a warning

Reply:
let me know what you think about my comment [here](https://github.com/evergreen-ci/evergreen/pull/4878#discussion_r679359493) about logging 404s